chainer/functions/array/reshape.py

```diff
@@ -51,12 +51,36 @@ def reshape(x, shape):
     """Reshapes an input variable without copy.
 
     Args:
-        x (~chainer.Variable): Input variable.
-        shape (tuple of ints): Target shape.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Input variable.
+        shape (:class:`tuple` of :class:`int` s):
+            The **size** of shape (**size** means the number of elements) must
```

Comment:
It seems to me the explanation of `size` could be misleading because we can also interpret it as the *length* of the `shape` (for example, some user could think the size of `(4, 5, 6)` is 3, rather than 4 * 5 * 6 = 120)